R/utilities-sensitivity-calculation.R

```diff
@@ -2,49 +2,221 @@
 
 #' Extract PK parameters dataframe from a list of `SimulationResults` objects
 #'
-#' @inheritParams .simulationResultsBatchToTimeSeriesDataFrame
+#' This function processes simulation result batches by parameterPath and
+#' extracts PK parameters into a dataframe.
+#'
+#' @param simulationResultsBatch List of simulation result batches.
+#' @param parameterPaths List of parameter paths corresponding to the simulation
+#' results.
+#' @param customOutputFunctions Optional list of custom output functions for
+#' user-defined PK analyses.
+#'
+#' @return A dataframe containing the PK parameters from all simulation results.
 #'
 #' @keywords internal
 #' @noRd
 .simulationResultsBatchToPKDataFrame <- function(simulationResultsBatch,
-                                                 parameterPaths) {
+                                                 parameterPaths,
+                                                 customOutputFunctions) {
   purrr::map2_dfr(
     .x = simulationResultsBatch,
     .y = parameterPaths,
-    .f = ~ .simulationResultsToPKDataFrame(.x, .y)
+    .f = ~ .simulationResultsToPKDataFrame(.x, .y, customOutputFunctions)
   )
 }
 
-#' Extract PK parameters dataframe from `Parameter` object
+#' Calculate PK parameters dataframe from simulation results
+#'
+#' This function calculates standard PK analyses from simulation results and
+#' converts them into a dataframe. If custom output functions are provided,
+#' it calculates user-defined PK analyses and integrates them with the standard
+#' PK data.
 #'
-#' @inheritParams .simulationResultsToTimeSeriesDataFrame
+#' @param simulationResults List of simulation results from which PK analyses
+#' are to be calculated.
+#' @param parameterPath Path to the parameter in the simulation results.
+#' @param customOutputFunctions Optional list of custom output functions for
```

Comment:
Can it work with anonymous functions ? And with `~` syntax  (e.g ~x+2)

Reply:
Yes, we already can pass it as a list of anonymous functions
`list("minmax" = function(y) min(y[y != 0]) / max(y))`

Purrr specific `~` syntax for anonymous functions could be implemented and translated using `purrr::as_mapper` or `rlang::as_function`

Reply:
If other function syntax are supported, corresponding tests should be added.

There also the shorthand `\(x)`: `\(x) x + 1` ➡️ `function(x) x + 1` (cf [this](https://stat.ethz.ch/R-manual/R-devel/library/base/html/function.html))

Reply:
So far only classical function syntax is supported